firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -709,23 +709,23 @@ public void testDoesNotRetryOnPermanentError() {
   public void testRetryOnAlreadyExistsError() {
     final FirebaseFirestore firestore = testFirestore();
     DocumentReference doc = firestore.collection("foo").document();
-    CountDownLatch latch = new CountDownLatch(2);
+    AtomicInteger retryCount = new AtomicInteger(0);
```

Comment:
As a rule, I always recommend to avoid naming a variable with a number of "retries", `retryCount` in this case. The reason is that it is not obvious whether or not it includes the first attempt, or if it's only counting the subsequent "retries". Please choose another name for this variable, such as `transactionCallbackCount` or `numTransactionCallbackInvocations` or something better.